cluster/manager/manager.go

```diff
@@ -1186,7 +1232,7 @@ func (c *ClusterManager) initListeners(
 		// already in cluster, check with listeners if it is OK to join
 		// this cluster.
 		for e := c.listeners.Front(); e != nil; e = e.Next() {
-			err := e.Value.(cluster.ClusterListener).CanNodeJoin(&c.selfNode, clusterInfo, nodeInitialized)
+			err := e.Value.(cluster.ClusterListener).CanNodeJoin(&c.selfNode, db, nodeInitialized)
```

Comment:
Send the latest ClusterInfo object (db) to listeners instead of an older clusterInfo object